File: 0623-add-one-row-to-tree/0623-add-one-row-to-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def addOneRow(self, root: Optional[TreeNode], val: int, depth: int) -> Optional[TreeNode]:
        if root is None:
            return None
        if depth == 1:
            temp = TreeNode(val, root, None)
            return temp
        elif depth == 2:
            left = TreeNode(val, root.left, None)
            right = TreeNode(val, None, root.right)
            root.left, root.right = left, right
            return root
        else:
            self.addOneRow(root.left, val, depth - 1)
            self.addOneRow(root.right, val, depth - 1)
            return root

File: 0623-add-one-row-to-tree/0623-add-one-row-to-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* addOneRow(TreeNode* root, int val, int depth) {
        if(root == NULL) return root;
        if(depth == 1){
            TreeNode *temp = new TreeNode(val, root, NULL);
            return temp;
        }
        else if(depth == 2)
        {
            TreeNode *left = new TreeNode(val, root->left, NULL);
            TreeNode *right = new TreeNode(val, NULL, root->right);
            root->left = left, root->right = right;
            return root;
        }
        else {
            addOneRow(root->left, val, depth-1);
            addOneRow(root->right, val, depth-1);
            return root;
        }
    }
};